scripts/lua2dox.lua

```diff
@@ -158,6 +158,18 @@ local function removeCommentFromLine(line)
   return line:sub(1, pos_comment - 1), line:sub(pos_comment)
 end
 
+--- Add possible pattern variations to a type. This includes arrays, and
+--- optional types.
+---
+---@param type string
+---@return string
+local function addPossibleTypeCaptureGroups(type)
+  -- [%[%]]* : match any dimensional array
+  -- [%?]* : match `?` option type (e.g., `string?`)
+  -- [|nil]* : match `|nil` option type (e.g., `string|nil`)
```

Comment:
This will match stuff like `||||`, `nnininil|||nni`.

Reply:
Ah, I was confusing lua patterns with regex.